swift/src/Ice/Incoming.swift

```diff
@@ -92,6 +98,22 @@ public final class Incoming {
         self.format = format
     }
 
+    public func setResultPromise<T>(_ p: Promise<T>,
+                                    _ cb: ((OutputStream, T) -> Void)? = nil) -> Promise<OutputStream> {
```

Comment:
I prefer the way Ice for Java does this, with two overloaded methods, one for Promise<T> with T != Void and a non-optional closure, and one for Promise<Void> and no closure at all.